src/java/com/puppetlabs/http/client/impl/JavaClient.java

```diff
@@ -493,4 +511,56 @@ public static Object coerceBodyType(InputStream body, ResponseBodyType as,
 
         return response;
     }
+
+    private static Timer.Context startTimer(MetricRegistry registry, HttpRequest request) {
+        if (registry != null) {
+            final RequestLine requestLine = request.getRequestLine();
+            final String name = MetricRegistry.name("puppetlabs.http-client", requestLine.getUri(),
```

Comment:
I wanted to get a PR up with all the metric plumbing and make sure that all was okay and then add in support for `metric-id` as a request option. The ticket for that is TK-317.

My thinking had been that `metric-id` would be optional, and if it wasn't specified use the raw URI and request method. However, I could see it maybe being useful to put all the requests without a metric-id under `unidentified-requests`?